tests/list-indentation.test.js

```diff
@@ -7,24 +7,1216 @@ const { markdownlint } = require('markdownlint').promises,
+         '',
+         '   * Includes body content and a [**Contact Information Chooser**](#TODO) for legal office',
+         '     contact information',
+      ]);
+   });
+
+   it('properly validates starting fenced code', async () => {
+      await testValidExample([
+         '',
+         '   * ul item',
+         '   * ul item',
+         '     continued paragraph',
+         '     ```',
+         '                  This content is not checked();',
+         '             not checked();',
+         '            not checked();',
+         '             not checked();',
+         '             not checked();',
+      ]);
+   });
+
+   it('should validate ol inside of a ul with no starting indentation (4 layers deep)', async () => {
+      await testValidExample([
+         '* ul item',
+         '  * ul item',
+         '    1. ol item',
+         '       * ul item',
+         '    2. ol item',
+      ]);
+   });
+
+   it('reports no errors for properly aligned links in lists', async () => {
+      await testValidExample([
+         '1. [This is a multiline link',
+         '   to test](#test-link)',
+         '2. [another link](#another-link)',
+      ]);
+   });
+
+   it('should validate list items inside of a blockquote', async () => {
+      await testValidExample([
+         '>> * ul item',
+         '>> * ul item',
+      ]);
+   });
+
+   it('should fail short indentation for a blockquote', async () => {
+      await testInvalidExample([
+         '>> ul item',
+         '   >> ul item',
+      ]);
+   });
+
+   it('should fail long indentation for an ol inside of an ul', async () => {
+      await testInvalidExample([
+         '* ul item',
+         '   * ul item',
+         '       1. ol item',
+         '          * ul item',
+         '       2. ol item',
+      ], { 'ul_indent': 3 });
+   });
+
+   it('should fail short indentation for a sub-list', async () => {
+      await testInvalidExample([
+         '   1. ol item',
+         '     * ul item',
+         '   2. ol item',
+      ]);
+   });
+
+   it('stars (bold or italics) should not throw off indent check', async () => {
+      await testValidExample([
+         '## test page',
+         '',
+         '   1. contact information',
+         '      1. contact information',
+         '         * contact information',
+         '            * Includes body content and a [**Contact Information Chooser**](#TODO) test',
+         '               * contact information',
+      ], { 'ul_indent': 3 });
+   });
+
+   it('should prioritize indent config setting over dynamic indent for child unordered lists', async () => {
+      await testValidExample([
+         '   1. ol item',
+         '      1. ol item',
+         '         * ul item',
+         '            * ul item',
+         '               * ul item',
+      ], { 'ul_indent': 3 });
+   });
+
+   it('should validate ol inside of a ul with starting indentation of 3 (3 layers deep)', async () => {
```

Comment:
This test could be moved inside of 'reports no errors for left-aligned list items'